drivers/adc/adc_stm32.c

```diff
@@ -233,11 +233,6 @@ static const uint32_t table_samp_time[] = {
 };
 #endif
 
-/* Bugfix for STM32G4 HAL */
-#if !defined(ADC_CHANNEL_TEMPSENSOR)
-#define ADC_CHANNEL_TEMPSENSOR ADC_CHANNEL_TEMPSENSOR_ADC1
-#endif
-
```

Comment:
I would not have called this a bug fix. This ignores the fact that the STM32G4 uses a different channel for sampling the internal temperature for ADCs 1 and 5.